src/main/kotlin/translations/TranslationFiles.kt

```diff
@@ -110,12 +111,48 @@ object TranslationFiles {
         element.delete()
     }
 
+    fun findTranslationKeyForText(context: PsiElement, text: String): String? {
+        val module = context.findModule()
+            ?: throw IllegalArgumentException("Cannot add translation for element outside of module")
+        var jsonVersion = true
+        if (!MinecraftSettings.instance.isForceJsonTranslationFile) {
+            val version =
+                context.mcVersion ?: throw IllegalArgumentException("Cannot determine MC version for element $context")
+            jsonVersion = version > MC_1_12_2
+        }
+
+        if (!jsonVersion) {
+            return null // todo: lang file support
+        }
+
+        val files = FileTypeIndex.getFiles(
+            if (jsonVersion) JsonFileType.INSTANCE else LangFileType,
+            GlobalSearchScope.moduleScope(module),
+        ).filter { getLocale(it) == TranslationConstants.DEFAULT_LOCALE }
+
+        for (file in files) {
+            val psiFile = PsiManager.getInstance(context.project).findFile(file) ?: continue
+            psiFile.apply {
+                if (jsonVersion) {
+                    this.findKeyForTextAsJson(text)?.let { return it }
+                } else {
+                    // todo: this.findKeyForTextAsLang(text)
+                }
+            }
```

Comment:
Not sure I'd go for the `apply` over just `psiFile.findKeyForTextAsJson(text)`

Reply:
[a135ee5](https://github.com/minecraft-dev/MinecraftDev/pull/2292/commits/a135ee54cdb3eb9d910636ea014f32906314292d)